src/include/elektra.h

```diff
@@ -0,0 +1,466 @@
+
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
+
+ELEKTRA_TAG_DECLARATIONS (int, Enum)
+
+// endregion Tags for built-in types
+
+// region Basics
+/**************************************
+ *
+ * Basics
+ *
+ **************************************/
+
+Elektra * elektraOpen (const char * application, KeySet * defaults, ElektraError ** error);
+void elektraClose (Elektra * elektra);
+
+// endregion Basics
+
+// region Error-Handling
+/**************************************
+ *
+ * Error-Handling
+ *
+ **************************************/
+
+void elektraFatalErrorHandler (Elektra * elektra, ElektraErrorHandler fatalErrorHandler);
+
+// endregion
+
+// region Helpers for code generation
+/**************************************
+ *
+ * Helpers for code generation
+ *
+ **************************************/
+
+Key * elektraFindKey (Elektra * elektra, const char * name, KDBType type);
+Key * elektraFindArrayElementKey (Elektra * elektra, const char * name, size_t index, KDBType type);
+void elektraFatalError (Elektra * elektra, ElektraError * fatalError);
+
+// endregion Helpers for code generation
+
+// region Getters
+/**************************************
+ *
+ * Getters
+ *
+ **************************************/
+
+const char * elektraGetValue (Elektra * elektra, const char * name);
+const char * elektraGetString (Elektra * elektra, const char * keyname);
+kdb_boolean_t elektraGetBoolean (Elektra * elektra, const char * keyname);
+kdb_char_t elektraGetChar (Elektra * elektra, const char * keyname);
+kdb_octet_t elektraGetOctet (Elektra * elektra, const char * keyname);
+kdb_short_t elektraGetShort (Elektra * elektra, const char * keyname);
+kdb_unsigned_short_t elektraGetUnsignedShort (Elektra * elektra, const char * keyname);
+kdb_long_t elektraGetLong (Elektra * elektra, const char * keyname);
+kdb_unsigned_long_t elektraGetUnsignedLong (Elektra * elektra, const char * keyname);
+kdb_long_long_t elektraGetLongLong (Elektra * elektra, const char * keyname);
+kdb_unsigned_long_long_t elektraGetUnsignedLongLong (Elektra * elektra, const char * keyname);
+kdb_float_t elektraGetFloat (Elektra * elektra, const char * keyname);
+kdb_double_t elektraGetDouble (Elektra * elektra, const char * keyname);
+
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
+
+kdb_long_double_t elektraGetLongDouble (Elektra * elektra, const char * keyname);
+
+#endif
+
+int elektraGetEnumInt (Elektra * elektra, const char * keyName);
+
+#ifdef __cplusplus
+#define elektraGetEnum(elektra, keyname, enumType) static_cast<enumType> (elektraGetEnumInt (elektra, keyname))
+#else
+#define elektraGetEnum(elektra, keyname, enumType) (enumType) elektraGetEnumInt (elektra, keyname)
+#endif
+
+// endregion Getters
+
+// region Setters
+/**************************************
+ *
+ * Setters
+ *
+ **************************************/
+
+void elektraSetValue (Elektra * elektra, const char * name, const char * value, KDBType type, ElektraError ** error);
```

Comment:
Also void *

Reply:
Same as for `elektraGetValue`, if we want to support binary data, we need to add a size parameter.

Reply:
See above.